base/dates/io.jl

```diff
@@ -152,7 +152,7 @@ function parse(x::AbstractString,df::DateFormat)
         throw(ArgumentError("Delimiter mismatch. Couldn't find first delimiter, \"$(df.slots[1].transition)\", in date string"))
     end
     periods = Period[]
-    extra = []
+    extra = Period[]
```

Comment:
@quinnj if we change the return statement later we can have parse return a `Period[]` in most cases:

Change:
```julia
return vcat(sort!(periods,rev=true,lt=periodisless), extra)
```
to
```julia
sort!(periods,rev=true,lt=periodisless)
if isempty(extra)
    return periods
else
    return vcat(periods, extra)
end
```

results in:
```julia
julia> eltype(Dates.parse("1942-12-25T01:23:45", Dates.DateFormat("yyyy-mm-ddTHH:MM:SS", "english")))
Base.Dates.Period

julia> eltype(Dates.parse("1942-12-25T01:23:45+0100", Dates.DateFormat("yyyy-mm-ddTHH:MM:SSzzz", "english")))
Any
```